Broken or Flaky test: AuthenticationTokenSecretManagerTest

**Test name(s)**
 - AuthenticationTokenSecretManagerTest.testVerifyPassword

**Describe the failure observed**

There seems to be some small probability that the AuthenticationTokenSecretManager will generate the same random password. I've seen the following error a few times, but it's hard to reproduce. Since it's calling into Hadoop code to generate things, this could depend on specific Hadoop versions.

```
[ERROR] Tests run: 10, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 11.485 s <<< FAILURE! - in org.apache.accumulo.server.security.delegation.AuthenticationTokenSecretManagerTest
[ERROR] org.apache.accumulo.server.security.delegation.AuthenticationTokenSecretManagerTest.testVerifyPassword  Time elapsed: 0.003 s  <<< FAILURE!
org.opentest4j.AssertionFailedError: Different tokens for the same user shouldn't have the same password ==> expected: <false> but was: <true>
	at org.junit.jupiter.api.AssertionFailureBuilder.build(AssertionFailureBuilder.java:151)
	at org.junit.jupiter.api.AssertionFailureBuilder.buildAndThrow(AssertionFailureBuilder.java:132)
	at org.junit.jupiter.api.AssertFalse.failNotFalse(AssertFalse.java:63)
	at org.junit.jupiter.api.AssertFalse.assertFalse(AssertFalse.java:36)
	at org.junit.jupiter.api.Assertions.assertFalse(Assertions.java:236)
	at org.apache.accumulo.server.security.delegation.AuthenticationTokenSecretManagerTest.testVerifyPassword(AuthenticationTokenSecretManagerTest.java:222)

```

**Additional context**

If I remember correctly, this code is part of the delegation token management stuff that is used with Kerberos-enabled clusters so that MapReduce and other similar tasks don't need a keytab for each individual job task. Most recently, I saw this on 3.0.0-SNAPSHOT in the main branch, but I've previously seen it in development on earlier branches as well.

--- a/server/base/src/test/java/org/apache/accumulo/server/security/delegation/AuthenticationTokenSecretManagerTest.java
+++ b/server/base/src/test/java/org/apache/accumulo/server/security/delegation/AuthenticationTokenSecretManagerTest.java
@@ -208,7 +208,7 @@ public class AuthenticationTokenSecretManagerTest extends WithTestNames {
     assertArrayEquals(password, secretManager.retrievePassword(id));
 
     // Make a second token for the same user
-    // Sleep for 1 millisecond to guarantee token is unique
+    // Briefly sleep to guarantee token is unique, since the token is based on the time
     Thread.sleep(100);
     Entry<Token<AuthenticationTokenIdentifier>,AuthenticationTokenIdentifier> pair2 =
         secretManager.generateToken(principal, cfg);

--- a/server/base/src/test/java/org/apache/accumulo/server/security/delegation/AuthenticationTokenSecretManagerTest.java
+++ b/server/base/src/test/java/org/apache/accumulo/server/security/delegation/AuthenticationTokenSecretManagerTest.java
@@ -209,7 +209,7 @@ public class AuthenticationTokenSecretManagerTest extends WithTestNames {
 
     // Make a second token for the same user
     // Sleep for 1 millisecond to guarantee token is unique
-    Thread.sleep(1);
+    Thread.sleep(100);
     Entry<Token<AuthenticationTokenIdentifier>,AuthenticationTokenIdentifier> pair2 =
         secretManager.generateToken(principal, cfg);
     Token<AuthenticationTokenIdentifier> token2 = pair2.getKey();

--- a/server/base/src/test/java/org/apache/accumulo/server/security/delegation/AuthenticationTokenSecretManagerTest.java
+++ b/server/base/src/test/java/org/apache/accumulo/server/security/delegation/AuthenticationTokenSecretManagerTest.java
@@ -208,6 +208,8 @@ public class AuthenticationTokenSecretManagerTest extends WithTestNames {
     assertArrayEquals(password, secretManager.retrievePassword(id));
 
     // Make a second token for the same user
+    // Sleep for 1 millisecond to guarantee token is unique
+    Thread.sleep(1);
     Entry<Token<AuthenticationTokenIdentifier>,AuthenticationTokenIdentifier> pair2 =
         secretManager.generateToken(principal, cfg);
     Token<AuthenticationTokenIdentifier> token2 = pair2.getKey();

